src/qutip_tensornetwork/core/data/network.py

```diff
@@ -672,3 +748,103 @@ def _match_edges_by_split(out_edges, in_edges):
     new_out_edges.reverse()
     new_in_edges.reverse()
     return new_out_edges, new_in_edges
+
+
+def _match_dimensions(edges, target_dims):
+    """This function splits the edges such that the dimensions of edges
+    matches target_dims. Only split of edges are allowed in order to match
+    target_dims.
+
+    Note
+    ----
+    The modifications to edges are performed in-place, which is why the
+    function is private as it is meant to be used only within network.
+
+    Examples
+    --------
+    >>> network_dim = [4]
+    >>> target_dims = [2, 2]
+    >>> array = np.random.random(network_dim)
+    >>> node = tn.Node(array)
+    >>> network = Network(node[:], [])
+    >>> _match_dimensions(network.in_edges, target_dims)
+
+    Since only splitting of edges are allowed, the next example will raise a
+    ValueError:
+    >>> network_dim = [2, 2]
+    >>> target_dims = [4]
+    >>> array = np.random.random(network_dim)
+    >>> node = tn.Node(array)
+    >>> network = Network(node[:], [])
+    >>> _match_dimensions(network.in_edges, target_dims) # ValueError
+    """
+
+    edges = edges[:]
+
+    target_dims = target_dims[:]
+    _target_dims = target_dims[:]  # This is kept for the error messages.
+
+    e_dims = [e.dimension for e in edges]
+
+    new_edges = []
+
+    if len(edges) == 0 and len(target_dims) == 0:
+        return _edges
```

Comment:
Should this just be `return edges`? `_edges` does not seem to be defined any more. We should add a test for the empty edges and target_dims case.